### DISCUSSION TITLE
[Support]: 0.14 Large number of cameras/filters pushes apply button off screen

### DISCUSSION THREAD

**Discussion by YBonline** [blakeblackshear/frigate#11156] [General Support]
### Describe the problem you are having

First off, absolutely amazing job on the new UI! The new interface is very slick, and sure to help Frigate become much more successful and makes it sooooo much more enjoyable to use.... I've been using it for a little while now, and its been absolutely amazing.

With that said, wanted to report a bug...
All of the dropdown menus need to be scrollable. For example, in reviews, the cameras and filters don't fit on my screen, so I can't hit the apply button, or see all my cameras. This is also an issue in camera groups, which uses a different style menu.

### Version

0.14

### Frigate config file

```yaml
NA
```

### Relevant log output

```shell
NA
```

### FFprobe output from your camera

```shell
NA
```

### Frigate stats

```json
NA
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

PCIe

### Network connection

Wired

### Camera make and model

NA

**Comment by NickM-27** [comment]
we were wondering if that would be the case, should be an easy fix

**Reply by NickM-27** [reply]
https://github.com/blakeblackshear/frigate/pull/11155

**Comment by YBonline** [comment]
Thanks, that fixes it! Mostly... when I add a camera group, AFTER I hit submit, there is some weird second scroll bar that is appearing and disappearing over and over again inside that dialog until the save is completed.

**Reply by NickM-27** [reply]
Interesting, will have to look at that

**Reply by YBonline** [reply]
I made a little video of it 

https://github.com/blakeblackshear/frigate/assets/11162258/eb37a106-5979-483e-8ba5-f26bbbec4154

**Comment by YBonline** [comment]
Also 2 more little UI issues:

1) On cameras with a 16:9 aspect ratio, the controls on the playback appear by default below the video player... however, they only appear when you hover on the video player, and if you hover off, to use the controls, they disappear before you get to them.
![Screenshot 2024-04-29 155317](https://github.com/blakeblackshear/frigate/assets/11162258/49c2858d-8dc5-4ec8-b50f-f1e00edde329)

2) On the timeline object, if you are on a desktop with a touchscreen, if you try to touch and drag the current time/red bar, it doesn't work. On mobile this works, and if you use a mouse, it works as expected, but not the touchscreen.

**Reply by NickM-27** [reply]
1. This is most likely a problem where your detect resolution and camera resolution are different aspect ratios. 16:9 cameras should not be appearing in this layout.

**Comment by YBonline** [comment]
> 1. This is most likely a problem where your detect resolution and camera resolution are different aspect ratios. 16:9 cameras should not be appearing in this layout.

Hmm, so yes, it does appear that the actual substream aspect ratio provided by the camera is in fact different then the high resolution aspect ratio.... and from what I can tell, this does not appear to be an option on the camera to make them the same... (which seems insane to me, but security cameras have the weirdest firmwares on board, so I guess no surprise). I don't have adequate hardware to re-encode my own substream given my number of cameras, so whats my best option there?

**Reply by NickM-27** [reply]
your best option is to manually set the detect resolution in frigate to match the aspect ratio of the cameras main stream. 

So for example if your main stream is 16:9 and the sub stream is 640x480 then just set the detect config to be

```
detect:
  width: 640
  height: 360
```

and that will have minimal if any additional resource usage

**Reply by YBonline** [reply]
and the stretching wouldn't impact detections significantly? I know I'm struggling slightly with my newer fisheye camera (although its not well trained as of yet, just with a near default model)

also, I'm curious, I'm admittedly entirely unfamiliar with Frigate's code base, but wouldn't it be better to just do the positioning on the client side based on the actual video ratio right when its playing? it doesn't seem like it'd have any disadvantage to doing it that way?

**Reply by NickM-27** [reply]
it is just more complicated to do because we do not have that information right away and also don't have it at the level that decides the layout

**Reply by NickM-27** [reply]
looks like Hawkeye worked on this (using the camera res directly instead of assuming detect matches)

**Comment by divemasterjm** [comment]
Hi, testing dev, if i can help ill be glad to do it,
when trying to send to frigate+ (in frigate+ tab) if i click on an snapshot the area of the image is way to big and i cant see the botton for sending or i can imagine is a person/is not a person.

**Reply by NickM-27** [reply]
an example screenshot would be helpful

**Reply by divemasterjm** [reply]
![Captura de pantalla 2024-04-30 161738](https://github.com/blakeblackshear/frigate/assets/46672242/3267ff91-fa46-4824-a371-18290b4f942a)

**Reply by NickM-27** [reply]
what is the resolution of the browser window? (if it is fullscreen, what is the monitor resolution?)

**Reply by divemasterjm** [reply]
browser window is full max and resolution is 1920x1080 with 150% scale

**Reply by divemasterjm** [reply]
if scale is 100% then i can see buttons, but if i resize browser window then i have same effect as before

**Comment by YBonline** [comment]
So just wanted to follow up... most of my issues are fixed... however:
In settings > masks/zones & motion tuner & debug, it still needs a scrollbar

Also touch drag events on the time on playback still don't work on desktop Chrome/tablets 

Everything else is resolved in the latest version

**Reply by hawkeye217** [reply]
A screenshot of the place that needs a scrollbar would be helpful.

**Reply by hawkeye217** [reply]
Actually, I'm all set - I just put up [this PR](https://github.com/blakeblackshear/frigate/pull/11241) to solve your issues.

**Reply by YBonline** [reply]
Thanks, that fixes that!

So I'm getting a couple more issues... both of them are very intermittent and demonstrated on this screenshot (they seem to happen about 10% of the time)
1. Sometimes the other cameras previews get really small, other times they get a scrollbar, as would be expected. I have not detected any pattern to the behavior.
2. Randomly the error about my camera being down (I have a network switch that failed and I'm waiting on a replacement) repeats a ton of times, other times, I only get one message about it, as expected. Again, with seemingly no pattern.
![Screenshot 2024-05-04 230529](https://github.com/blakeblackshear/frigate/assets/11162258/583dabaa-89e0-416d-998d-14eac6082cd9)

Also I'm really digging the alerts/detections as the background on the timeline, that looks awesome, great improvement!
